### ISSUE TITLE
Allow declaring "version groups" for packages

### ISSUE THREAD

**Issue by epage** [MarcoIeni/release-plz#1023]
## Motivations

When dealing with proc-macros, its common to have the package that re-exports it to have its version in lockstep with the proc macro package.  You can use `workspace.package.version` for this but that doesn't scale if you have other content in the repo (e.g. `clap`).

Note: just because they are in the group, doesn't mean all members have to release.

- Would you like to implement this feature? maybe

## Solution

`cargo-release` generalized `workspace.package.version` support into the package config field [`shared-version`](https://github.com/crate-ci/cargo-release/blob/master/docs/reference.md#config-fields) which can either be `true` or be a named group (with workspace versions being `"workspace"`)

**Comment by MarcoIeni** [comment]
Ok, so this feature allows groups of packages to share the same version, that can be different from `workspace.package.version`.

Sounds good and PR is welcome!

We can add the `shared-version` field to the [`package`](https://release-plz.ieni.dev/docs/config#the-package-section) section of the config file.

One note: I don't think that `true` is needed, because you could set `version.workspace = true` in the `Cargo.toml`, right?

**Comment by epage** [comment]
Yes, the `bool` state isn't strictly needed.  In part, that came from how it evolved in cargo-release.
